lib/data_hygiene/document_reslugger.rb

```diff
@@ -0,0 +1,72 @@
+module DataHygiene
+  # Reslugs a document to new_slug.
+  #
+  # When run, the following happens:
+  #
+  #  - changes the documents slug
+  #  - reindexes the document with it's new slug
+  #  - republishes the document to Publishing API (automatically handles the redirect)
+  #
+  class DocumentReslugger
+    attr_reader :document, :published_edition, :current_user, :new_slug
+
+    def initialize(document, published_edition, current_user, new_slug)
+      @document = document
+      @published_edition = published_edition
+      @current_user = current_user
+      @new_slug = new_slug
+    end
+
+    def run!
+      add_errors_if_invalid
+      return false if document.errors.present?
+
+      remove_from_search_index
+      save_document
+      republish_document
+      add_to_search_index
+      create_editorial_remark
+      true
+    end
+
+  private
+
+    def add_errors_if_invalid
+      document.errors.add(:slug, "is blank") and return if new_slug.blank?
+
+      document.errors.add(:slug, "must be unique") and return if new_slug_is_a_duplicate
+
+      document.errors.add(:slug, "is invalid") unless new_slug.starts_with?("/") && new_slug !~ %r{//} && new_slug !~ %r{./\z}
```

Comment:
I've just tried this task and I think we want to ensure the slug doesn't start with `/`, else we add that into the slug too.